verifiers/internal/gha/builder.go

```diff
@@ -175,39 +190,309 @@ func verifyTrustedBuilderRef(id *WorkflowIdentity, ref string) error {
+	if err != nil {
+		return nil, err
+	}
+	if err := validateClaimsEqual(deprecatedSourceSha1, sourceSha1); err != nil {
+		return nil, err
+	}
+	if sourceSha1 == "" {
+		sourceSha1 = deprecatedSourceSha1
+	}
+
+	// 1.3.6.1.4.1.57264.1.14 | Source Repository Ref
+	// https://github.com/sigstore/fulcio/blob/main/docs/oid-info.md#13614157264114--source-repository-ref
+	sourceRef, err := getExtension(cert, "1.3.6.1.4.1.57264.1.14", true)
+	if err != nil {
+		return nil, err
+	}
+
+	// 1.3.6.1.4.1.57264.1.15 | Source Repository Identifier
+	// https://github.com/sigstore/fulcio/blob/main/docs/oid-info.md#13614157264115--source-repository-identifier
+	sourceID, err := getExtension(cert, "1.3.6.1.4.1.57264.1.15", true)
+	if err != nil {
+		return nil, err
+	}
+
+	// 1.3.6.1.4.1.57264.1.17 | Source Repository Owner Identifier
+	// https://github.com/sigstore/fulcio/blob/main/docs/oid-info.md#13614157264117--source-repository-owner-identifier
+	sourceOwnerID, err := getExtension(cert, "1.3.6.1.4.1.57264.1.17", true)
+	if err != nil {
+		return nil, err
+	}
+
+	// 1.3.6.1.4.1.57264.1.19 | Build Config Digest
+	// https://github.com/sigstore/fulcio/blob/main/docs/oid-info.md#13614157264119--build-config-digest
+	buildConfigSha1, err := getExtension(cert, "1.3.6.1.4.1.57264.1.19", true)
+	if err != nil {
+		return nil, err
+	}
+	// NOTE: sourceSha1 is *not* deprecated.
+	if deprecatedSourceSha1 != "" {
+		if err := validateClaimsEqual(deprecatedSourceSha1, buildConfigSha1); err != nil {
+			return nil, err
+		}
+	} else if sourceSha1 != "" {
+		if err := validateClaimsEqual(sourceSha1, buildConfigSha1); err != nil {
+			return nil, err
+		}
+	}
+
+	// 1.3.6.1.4.1.57264.1.18 | Build Config URI
+	// https://github.com/sigstore/fulcio/blob/main/docs/oid-info.md#13614157264118--build-config-uri
+	var buildConfigPath string
+	buildConfigURI, err := getExtension(cert, "1.3.6.1.4.1.57264.1.18", true)
+	if err != nil {
+		return nil, err
+	}
+	if buildConfigURI != "" {
+		parts := strings.Split(buildConfigURI, "@")
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("%w: %v",
+				serrors.ErrorInvalidFormat, buildConfigURI)
+		}
+		prefix := fmt.Sprintf("https://github.com/%v/", sourceRepository)
+		if !strings.HasPrefix(parts[0], prefix) {
+			return nil, fmt.Errorf("%w: prefix: %v",
+				serrors.ErrorInvalidFormat, parts[0])
+		}
+		buildConfigPath = strings.TrimPrefix(parts[0], prefix)
+	}
+
+	// 1.3.6.1.4.1.57264.1.21 | Run Invocation URI
+	// https://github.com/sigstore/fulcio/blob/main/docs/oid-info.md#13614157264121--run-invocation-uri
+	runURI, err := getExtension(cert, "1.3.6.1.4.1.57264.1.21", true)
+	if err != nil {
+		return nil, err
+	}
+	runID := strings.TrimPrefix(runURI, fmt.Sprintf("https://github.com/%s/actions/runs/", sourceRepository))
+
+	// Subject path.
+	if !strings.HasPrefix(cert.URIs[0].Path, "/") {
+		return nil, fmt.Errorf("%w: %s", serrors.ErrorInvalidFormat, cert.URIs[0].Path)
+	}
```

Comment:
nit: save the trimmed path to a variable close to where it was validated.
```suggestion
	// Subject path.
	if !strings.HasPrefix(cert.URIs[0].Path, "/") {
		return nil, fmt.Errorf("%w: %s", serrors.ErrorInvalidFormat, cert.URIs[0].Path)
	}
	// Remove the starting '/'
	workflowRef = cert.URIs[0].Path[1:]
```